sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,344 @@
+        dataset.data : numpy array of shape (494021, 41)
+            Each row corresponds to the 41 features in the dataset.
+        dataset.target : numpy array of shape (494021,)
+            Each value corresponds to one of the 21 attack types or to the
+            label 'normal.'.
+        dataset.DESCR : string
+            Description of the kddcup99 dataset.
+
+    """
+
+    data_home = get_data_home(data_home=data_home)
+    if sys.version_info[0] == 3:
+        # The zlib compression format use by joblib is not compatible when
+        # switching from Python 2 to Python 3, let us use a separate folder
+        # under Python 3:
+        dir_suffix = "-py3"
+    else:
+        # Backward compat for Python 2 users
+        dir_suffix = ""
+    if percent10:
+        kddcup_dir = join(data_home, "kddcup99_10" + dir_suffix)
+    else:
+        kddcup_dir = join(data_home, "kddcup99" + dir_suffix)
+    samples_path = join(kddcup_dir, "samples")
+    targets_path = join(kddcup_dir, "targets")
+    available = exists(samples_path)
+
+    if download_if_missing and not available:
+        _mkdirp(kddcup_dir)
+        URL_ = URL10 if percent10 else URL
+        logger.warning("Downloading %s" % URL_)
+        f = BytesIO(urlopen(URL_).read())
+
+        dt = [('duration', int),
+              ('protocol_type', 'S4'),
+              ('service', 'S11'),
+              ('flag', 'S6'),
+              ('src_bytes', int),
+              ('dst_bytes', int),
+              ('land', int),
+              ('wrong_fragment', int),
+              ('urgent', int),
+              ('hot', int),
+              ('num_failed_logins', int),
+              ('logged_in', int),
+              ('num_compromised', int),
+              ('root_shell', int),
+              ('su_attempted', int),
+              ('num_root', int),
+              ('num_file_creations', int),
+              ('num_shells', int),
+              ('num_access_files', int),
+              ('num_outbound_cmds', int),
+              ('is_host_login', int),
+              ('is_guest_login', int),
+              ('count', int),
+              ('srv_count', int),
+              ('serror_rate', float),
+              ('srv_serror_rate', float),
+              ('rerror_rate', float),
+              ('srv_rerror_rate', float),
+              ('same_srv_rate', float),
+              ('diff_srv_rate', float),
+              ('srv_diff_host_rate', float),
+              ('dst_host_count', int),
+              ('dst_host_srv_count', int),
+              ('dst_host_same_srv_rate', float),
+              ('dst_host_diff_srv_rate', float),
+              ('dst_host_same_src_port_rate', float),
+              ('dst_host_srv_diff_host_rate', float),
+              ('dst_host_serror_rate', float),
+              ('dst_host_srv_serror_rate', float),
+              ('dst_host_rerror_rate', float),
+              ('dst_host_srv_rerror_rate', float),
+              ('labels', 'S16')]
```

Comment:
Can't we have float everywhere, except for the labels which should be of dtype object?